routing/result_interpretation.go

```diff
@@ -857,6 +876,23 @@ func (i *interpretedResult) failBlindedRoute(rt *mcRoute) {
 	i.pairResults[pair] = failPairResult(amt)
 }
 
+// markBlindedRouteSuccess marks the hops of the blinded route AFTER the
+// introduction node as successful.
+//
+// NOTE: The introIdx must be the index of the first hop of the blinded route
+// AFTER the introduction node.
+func (i *interpretedResult) markBlindedRouteSuccess(rt *mcRoute, introIdx int) {
+	// For blinded hops we do not have the forwarding amount so we take the
+	// minimal amount which went through the route by looking at the last
+	// hop.
+	successAmt := rt.hops.Val[len(rt.hops.Val)-1].amtToFwd.Val
```

Comment:
just a comment: wanted to note that using the receiver amount won't help the apriori probability estimator for the other hops than the last one, because it doesn't make any conclusions for amounts larger than the last success amount. The probability will stay at the default probability for those hops. So in the test case if we succeeded with 58 sat, we will compute a probability of 0.6 * 1 in the next pathfinding round. If we would use the amount that enters the intro node, we'd compute a higher probability of 1 * 1 and that effect gets stronger the longer the route is. I think it's ok to go with the current approach, just wanted to mention this detail.

Reply:
As a clarification of the comment, lets say one has real pathfinding amounts of 10000-10001-10002 (from receiver to the payer, with fees), if this is all in a blinded route, we would set the success amount to 10000 for all those hops. In the next pathfinding iteration, also delivering 10000 sat would give then a probability of 1.0*0.6*0.6, because we only assign a full probability of 1, if the amount that we try to send over the channel is above the last success amount, which it isn't for 10001 and 10002. The bimodal model can extrapolate from 10000 to 10001/2, but the apriori one can't.